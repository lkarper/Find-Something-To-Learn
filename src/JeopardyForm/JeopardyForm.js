import React, { Component } from 'react';
import AppContext from '../AppContext/AppContext';

class JeopardyForm extends Component {

    static contextType = AppContext;

    state = {
        numberQuestions: 10,
    };

    setNumberQuestions = (numberQuestions) => {
        this.setState({
            numberQuestions
        });
    }

    fetchJeoTrivia = (event) => {
        event.preventDefault();
        const url = `http://jservice.io/api/random?count=${this.state.numberQuestions}`;
        fetch(url)
            .then(response => {
                if (response.ok) {
                    return response.json();
                } else {
                    console.log(response);
                    throw new Error(response.statusText);
                }
            })
            .then(responseJson => {
                console.log(responseJson);
                this.context.handleNewQuestionsJeo(responseJson);
            })
            .catch(error => console.log('error', error));
    }

    render() {
        return (
            <form onSubmit={e => this.fetchJeoTrivia(e)}>
                <label htmlFor="num-qs2">How many questions would you like?</label>
                <input 
                    id="num-qs2" 
                    type="number" 
                    min="5" 
                    max="100" 
                    value={this.state.numberQuestions}
                    required
                    onChange={e => this.setNumberQuestions(e.target.value)}
                    />
                <button type="submit">Let's Play Jeopardy!</button>
            </form>
        );
    }
}

export default JeopardyForm;