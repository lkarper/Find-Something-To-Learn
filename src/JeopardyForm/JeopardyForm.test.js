import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import JeopardyForm from './JeopardyForm';

describe('JeopardyForm component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<JeopardyForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = shallow(<JeopardyForm />);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
});