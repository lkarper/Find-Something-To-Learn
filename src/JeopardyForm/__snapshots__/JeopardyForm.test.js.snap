// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JeopardyForm component renders the UI as expected 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="num-qs2"
    >
      How many questions would you like?
    </label>
    <input
      id="num-qs2"
      max="100"
      min="5"
      onChange={[Function]}
      required={true}
      type="number"
      value={10}
    />
    <button
      type="submit"
    >
      Let's Play Jeopardy!
    </button>
  </form>
</div>
`;
